Makefile

```diff
@@ -33,26 +33,52 @@ venv: $(VENV_ACTIVATE)    ## Create a new (empty) virtual environment
 freeze:                   ## Run pip freeze -l in the virtual environment
 	@$(VENV_RUN); pip freeze -l
 
+pip-tools: venv
+	$(VENV_RUN); $(PIP_CMD) install --upgrade pip-tools
+
+upgrade-basic-reqs: pip-tools
+	$(VENV_RUN); pip-compile --upgrade --strip-extras -o requirements-basic.txt pyproject.toml
+
+upgrade-runtime-reqs: pip-tools
+	$(VENV_RUN); pip-compile --upgrade --extra runtime -o requirements-runtime.txt pyproject.toml
+
+upgrade-test-reqs: pip-tools
+	$(VENV_RUN); pip-compile --upgrade --extra test -o requirements-test.txt pyproject.toml
+
+upgrade-dev-reqs: pip-tools
+	$(VENV_RUN); pip-compile --upgrade --extra dev,test,runtime -o requirements-dev.txt pyproject.toml
+
+upgrade-dev-types-reqs: pip-tools
+	$(VENV_RUN); pip-compile --upgrade --extra typehint -o requirements-typehint.txt pyproject.toml
+
+upgrade-s3-reqs: pip-tools
+	$(VENV_RUN); pip-compile --upgrade --extra base-runtime -o requirements-base-runtime.txt pyproject.toml
```

Comment:
Why would I need these individual targets? I always have to update all of them, right?